subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/internal/precompiled/PrecompileGroovyScriptsTask.java

```diff
@@ -0,0 +1,299 @@
+                                       Provider<Directory> javaSourceDependencyClasses,
+                                       Provider<Directory> groovySourceDependencyClasses) {
+        this.scriptCompilationHandler = scriptCompilationHandler;
+        this.compileOperationFactory = compileOperationFactory;
+        this.fileSystemOperations = fileSystemOperations;
+        this.scriptRunnerFactory = scriptRunnerFactory;
+        this.serviceRegistry = serviceRegistry;
+
+        this.classLoaderScope = classLoaderScopeRegistry.getCoreAndPluginsScope();
+
+        this.scriptPlugins = scriptPlugins;
+        this.classpath = classpath;
+
+        DirectoryProperty buildDir = project.getLayout().getBuildDirectory();
+        this.intermediatePluginBlockClassesDir = buildDir.dir("groovy-dsl-plugins/plugin-blocks");
+        this.intermediatePluginClassesDir = buildDir.dir("groovy-dsl-plugins/work/classes");
+        this.intermediatePluginMetadataDir = buildDir.dir("groovy-dsl-plugins/work/metadata");
+
+        this.precompiledGroovyScriptsOutputDir.set(buildDir.dir("groovy-dsl-plugins/output/plugin-classes"));
+        this.pluginAdapterSourcesOutputDir.set(buildDir.dir("groovy-dsl-plugins/output/adapter-src"));
+        this.pluginAdapterClassesOutputDir.set(buildDir.dir("groovy-dsl-plugins/output/adapter-classes"));
+
+        this.javaSourceDependencyClasses = javaSourceDependencyClasses;
+        this.groovySourceDependencyClasses = groovySourceDependencyClasses;
+    }
+
+    @PathSensitive(PathSensitivity.RELATIVE)
+    @InputFiles
+    Set<File> getScriptFiles() {
+        return scriptPlugins.stream().map(p -> p.getSource().getResource().getFile()).collect(Collectors.toSet());
+    }
+
+    @PathSensitive(PathSensitivity.RELATIVE)
+    @InputFiles
+    Provider<Directory> getJavaSourceDependencyClasses() {
+        return javaSourceDependencyClasses;
+    }
+
+    @PathSensitive(PathSensitivity.RELATIVE)
+    @InputFiles
+    Provider<Directory> getGroovySourceDependencyClasses() {
+        return groovySourceDependencyClasses;
+    }
+
+    @Classpath
+    public FileCollection getClasspath() {
+        return classpath;
+    }
+
+    @OutputDirectory
+    Provider<File> getPrecompiledGroovyScriptsOutputDir() {
+        return precompiledGroovyScriptsOutputDir.getAsFile();
+    }
+
+    @OutputDirectory
+    DirectoryProperty getPluginAdapterSourcesOutputDir() {
+        return pluginAdapterSourcesOutputDir;
+    }
+
+    @Internal
+    Provider<File> getAdapterClassesOutputDir() {
+        return pluginAdapterClassesOutputDir.getAsFile();
+    }
+
+    @TaskAction
+    void compileScripts() {
+        FileCollection compileClasspath = classpath.plus(project.files(javaSourceDependencyClasses, groovySourceDependencyClasses));
+        ClassLoader compileClassLoader = new URLClassLoader(DefaultClassPath.of(compileClasspath).getAsURLArray(), classLoaderScope.getLocalClassLoader());
```

Comment:
Hmm, this doesn't look right to me (including the classloader of the classloader scope).  I think this is something that would all be driven by the dependencies of the plugin.  I think this ties compilation to the version of Gradle that's being used to compile the script.

Reply:
Whoa, I removed it and everything looked fine until I ran instant execution tests which turned out to be completely broken, complaining about:
```
  bar_cczf400jz7r8iu9kt0pmg15il: AST transformation implementation classes must be annotated with org.codehaus.groovy.transform.GroovyASTTransformation. org.codehaus.groovy.transform.BaseScriptASTTransformation lacks this annotation.
```